### ISSUE TITLE
Error comparing type T with Nullable<T>

### ISSUE THREAD

**Issue by dlahmad** [davideicardi/DynamicExpresso#123]
In the current implementation there is bug with comparing a type with its nullable version. Can this be fixed?

Best regards
Stephen

```
using DynamicExpresso;
using System;
using System.Globalization;

namespace TestApp1
{
    public class ExampleObject
    {
        public DateTimeOffset? SpecialDate;
    }

    class Program
    {

        static readonly Func<string, DateTimeOffset> Date = s => DateTimeOffset.Parse(s, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind);

        static void Main(string[] args)
        {
            var interpreter = new Interpreter()
                .Reference(typeof(DateTimeOffset))
                .SetFunction("Date", Date);


            // In Code this works
            var o = new ExampleObject()
            {
                SpecialDate = Date("2020-09-29")
            };

            var res = o.SpecialDate < Date("2020-09-30");

            // For parsing this does not work: The binary operator LessThan is not defined for the types 'System.Nullable`1[System.DateTimeOffset]' and 'System.DateTimeOffset'.'
            var exp = interpreter.ParseAsExpression<Func<ExampleObject, bool>>("o.SpecialDate < Date(\"2020-09-30\")", "o");
        }
    }
}

```

**Comment by davideicardi** [comment]
Thanks fore reporting. This is probably a bug.

I think that there should be already a PR available, see #122 . The problem is that this PR is not well formatted. If you have time can you try to run your code with that branch and create a new PR with the correct formatting (space vs tab)?

Otherwise I will try to investigate this in the future...

**Comment by dlahmad** [comment]
@davideicardi Could you then look at my PR # 124?
